extensions/container-image/container-image-openshift/deployment/src/main/java/io/quarkus/container/image/openshift/deployment/OpenshiftProcessor.java

```diff
@@ -234,6 +237,36 @@ public void openshiftRequirementsNative(OpenshiftConfig openshiftConfig,
         }
     }
 
+    @BuildStep(onlyIf = { OpenshiftBuild.class })
+    public void configureExternalRegistry(ApplicationInfoBuildItem applicationInfo,
+            OpenshiftConfig openshiftConfig,
+            ContainerImageInfoBuildItem containerImageInfo,
+            BuildProducer<DecoratorBuildItem> decorator) {
+        containerImageInfo.registry.ifPresent(registry -> {
+            String imagePushSecret = openshiftConfig.imagePushSecret.orElse(applicationInfo.getName() + "-push-secret");
+            if (openshiftConfig.imagePushSecret.isPresent()) {
+                // Process
+            } else if (containerImageInfo.username.isPresent() && containerImageInfo.password.isPresent()) {
```

Comment:
I think we should also support binding a file with the docker configuration, and also bind the local dockerconfig json file as we do for Tekton.
The `AddDockerConfigJsonSecretDecorator` decorator already provides this capability.

Reply:
I have mixed feelings about it.